.github/workflows/w3sperjs_ci.yml

```diff
@@ -0,0 +1,84 @@
+name: w3sper.js CI
+
+on:
+  workflow_dispatch:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - ready_for_review
+
+jobs:
+  # JOB to run change detection
+  changes:
+    runs-on: ubuntu-latest
+    # Required permissions
+    permissions:
+      pull-requests: read
+    # Set job outputs to values from filter step
+    outputs:
+      run-ci: ${{ steps.filter.outputs.run-ci }}
+    steps:
+      # For pull requests it's not necessary to checkout the code
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            run-ci:
+              - 'execution-core/**'
+              - 'wallet-core/**'
+              - 'w3sper.js/**'
+              - '.github/workflows/w3sperjs_ci.yml'
+
+  deno_lint_and_fmt_check:
+    needs: changes
+    # Skip lint check, see issue #2891
+    if: false
+    name: Deno lint check
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Setup deno
+        uses: denoland/setup-deno@v2
+        with: 
+          deno-version: v2.x
+      
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Run deno lint        
+        run: |
+          cd w3sper.js
+          deno lint
+          deno fmt --check
+
+  make_wasm_and_w3sper_test:
```

Comment:
Never use `_and_` for naming. If this is the main entry point for all the steps, just find a name that suits. For the time being we can just call it `w3sper_test` or something similar.